snap/container.go

```diff
@@ -33,7 +33,10 @@ import (
 // Container is the interface to interact with the low-level snap files
 type Container interface {
 	// ReadFile returns the content of a single file from the snap.
-	ReadFile(relative string) (content []byte, err error)
+	ReadFile(relative string) ([]byte, error)
+
+	// ReadDir returns the content of a single directory inside the snap.
+	ReadDir(path string) ([]os.FileInfo, error)
```

Comment:
mmh, we don't do those checks for apps either atm though, I agree that directory vs regular file is a bit borderline here though